README.md

```diff
@@ -15,251 +18,331 @@ The role can be used to configure:
+#### `state: down`
 
-* For initscripts it results in the deletion of the ifcfg file. Usually that
-  has no side-effect, unless some component is watching the sysconfig directory.
+- For `NetworkManager`,  it corresponds to `nmcli connection id {{name}} down`.
 
-#### Example
+- For `initscripts`, it corresponds to call `ifdown {{name}}`.
 
-```yaml
-network_connections:
-  - name: eth0
-    #persistent_state: present  # default
-    type: ethernet
-    autoconnect: yes
-    mac: 00:00:5e:00:53:5d
-    ip:
-      dhcp4: yes
-```
+You can deactivate a connection profile, even if is currently not active. As a consequence, `state: down` always changes the system.
 
-Above example creates a new connection profile or ensures that it is present
-with the given configuration. It implies the `persistent_state` setting to be
-`present`.
+Note that if the `state` option is unset, the connection profile’s runtime state will not be changed.
 
-Valid values for `type` are:
 
-  - `bond`
-  - `bridge`
+### `persistent_state`
+
+The `persistent_state` option identifies if a connection profile is in a persistent state. The `persistent_state` option can be set to the following values:
+
+*   `present` (default)
+
+    Note that if `persistent_state` is `present` and the connection profile contains
+    the `type` option, the profile will be created or updated. If the connection profile is
+    incomplete (no `type` option), the behavior is undefined. Also, the `present` value
+    does not directly result in a change in the network configuration. If the `state` option
+    is not set to `up`, the profile is only created or modified, not activated.
+
+*   `absent`
+
+    The `absent` value ensures that the profile is not present on the target host. For
+    example, if a profile with `name` `eth0` exists, it will be deleted. In this case:
+
+    - `NetworkManager` deletes all connection profiles with the corresponding `connection.id`.
+      Deleting a profile usually does not change the current networking configuration, unless
+      the profile was currently activated on a device. Deleting the currently
+      active connection profile disconnects the device. That makes the device eligible
+      to autoconnect another connection (for more details, see [rh#1401515](https://bugzilla.redhat.com/show_bug.cgi?id=1401515)).
+
+    - ` initscripts` deletes the ifcfg file in most cases with no impact on the system unless a component relies on the sysconfig directory.
+
+**Note**: For profiles that only contain a `state` option, the `network` role only activates
+or deactivates the connection without changing its configuration.
+
+
+### `type`
+
+The `type` option can be set to the following values:
+
   - `ethernet`
-  - `infiniband`
-  - `macvlan`
+  - `bridge`
+  - `bond`
   - `team`
   - `vlan`
+  - `macvlan`
+  - `infiniband`
+
+#### `type: ethernet`
+
+The `type:ethernet` should be specified as a dictionary with the following
```

Comment:
If the type is ethernet, then there can be an extra `ethernet` dictionary with the following settings.